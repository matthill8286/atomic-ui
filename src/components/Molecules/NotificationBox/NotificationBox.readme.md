## NotificationBox molecule

A component to show errors/notifications or to be used as a tooltip.
