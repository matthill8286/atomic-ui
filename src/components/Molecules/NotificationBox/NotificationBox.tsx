import React from 'react'
import { Card } from '@/components/Atoms/Card'
import { Icon } from '@/components/Atoms/Icon'
import { Link } from '@/components/Atoms/Link'
import { CopyText } from '@/components/Atoms/Typography/CopyText'
import { Padding } from '@/types/theme'
import {
  NotificationBoxProps,
  NotificationContentSort,
  NotificationLink,
} from './NotificationBox.interface'
import {
  StyledArrow,
  StyledBodyHtml,
  StyledButton,
  StyledButtonDiv,
  StyledCopyTextDiv,
  StyledLinkDiv,
  StyledNotificationboxWrapper,
  StyledNotificationClose,
  StyledNotificationHeadline,
} from './NotificationBoxStyled'
import {
  OtherArrow,
  OtherCheckmarkCircleOutlined,
  OtherClear,
  OtherClearCircleOutlined,
  OtherInfoOutlined,
  OtherSwapHorizontal,
} from '@matthill8286/atomic-icon-library'

export const NotificationBox = React.forwardRef<HTMLDivElement, NotificationBoxProps>(
  (
    {
      body,
      bodyFontSize = 'sm',
      bodyMargin,
      buttonLayout = 'column',
      buttons,
      cardShape = 'rounded-small',
      hasTitleIcon,
      isTitleFontBold = true,
      titleIconSelfAlign,
      isClosable,
      rootPosition,
      maxWidth,
      links,
      onClose,
      title,
      tooltip,
      type,
      sort = NotificationContentSort.LinksButtons,
      alignLinks = 'left',
      ...otherProps
    },
    ref
  ) => {
    const renderIcon = (type): React.ReactElement => {
      switch (type) {
        case 'info':
          return <OtherInfoOutlined />
        case 'success':
          return <OtherCheckmarkCircleOutlined />
        case 'alert':
          return <OtherInfoOutlined />
        case 'error':
          return <OtherClearCircleOutlined />
        case 'hot':
          return <OtherSwapHorizontal />
        default:
          return <OtherInfoOutlined />
      }
    }

    // Originally "iconLeft" was not settable
    // Thus to ensure backwards compatibility and also allow to set no "iconLeft" to be rendered,
    // null will return undefined and undefined will return a default element.
    const getIconLeft = (iconLeft?: React.ReactElement | null): React.ReactElement | undefined => {
      if (iconLeft === null) return undefined
      if (iconLeft === undefined)
        return (
          <Icon color="primary">
            <OtherArrow />
          </Icon>
        )
      return iconLeft
    }

    const renderLinks = () =>
      Boolean(links?.length) && (
        <StyledLinkDiv alignLinks={alignLinks}>
          {/* TypeScript cannot infer that 'links' cannot be 'undefined' at this point (yet) */}
          {(links as NotificationLink[]).map(({ label, iconLeft, ...linkProps }) => (
            <Link
              iconLeft={getIconLeft(iconLeft)}
              color="white"
              decorationColor="white"
              key={label}
              scale="large"
              {...linkProps}>
              {label}
            </Link>
          ))}
        </StyledLinkDiv>
      )

    const arrowPos = tooltip?.arrowPosition.split('-')
    const padding = tooltip ? ({ mobile: 'sm', tablet: 'md', top: 'md' } as Padding) : 'md'

    return (
      <StyledNotificationboxWrapper
        data-test="notification-box"
        tooltip={tooltip}
        type={type}
        rootPosition={rootPosition}
        maxWidth={maxWidth}
        ref={ref}
        {...otherProps}>
        <StyledNotificationClose onClick={onClose}>
          <Icon width={16} height={16} color={'white'}>
            {isClosable && <OtherClear />}
          </Icon>
        </StyledNotificationClose>
        {arrowPos && <StyledArrow arrowLeft={tooltip?.arrowleft} arrowPosition={arrowPos} />}
        <Card elevation={1} padding={padding} shape={cardShape} surface="black">
          {title && hasTitleIcon && (
            <StyledNotificationHeadline>
              <Icon width={16} height={16} color="white" alignSelf={titleIconSelfAlign}>
                {renderIcon(type)}
              </Icon>
              <CopyText
                data-test="notification-box-title"
                color="white"
                tag="div"
                bold={isTitleFontBold}>
                {title}
              </CopyText>
            </StyledNotificationHeadline>
          )}
          {body && (
            <StyledCopyTextDiv margin={bodyMargin}>
              <CopyText
                data-test="notification-box-body"
                color="white"
                fontSize={bodyFontSize}
                margin={bodyMargin}>
                {
                  <StyledBodyHtml
                    dangerouslySetInnerHTML={{
                      __html: body,
                    }}
                  />
                }
              </CopyText>
            </StyledCopyTextDiv>
          )}

          {sort === NotificationContentSort.LinksButtons && renderLinks()}

          {buttons && buttons.length > 0 && (
            <StyledButtonDiv buttonLayout={buttonLayout}>
              {buttons.map(({ actionBtnLabel, actionType = 'inverted', ...buttonProps }) => (
                <StyledButton
                  actionType={actionType}
                  fullWidth
                  key={actionBtnLabel}
                  size="sm"
                  {...buttonProps}>
                  {actionBtnLabel}
                </StyledButton>
              ))}
            </StyledButtonDiv>
          )}
          {sort === NotificationContentSort.ButtonsLinks && renderLinks()}
        </Card>
      </StyledNotificationboxWrapper>
    )
  }
)
