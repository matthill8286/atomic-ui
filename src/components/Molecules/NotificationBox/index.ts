export * from './NotificationBox'
export * from './NotificationBox.interface'
