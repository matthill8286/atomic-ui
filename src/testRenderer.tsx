import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
