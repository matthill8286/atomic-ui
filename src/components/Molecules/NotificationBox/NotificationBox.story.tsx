import { action } from '@storybook/addon-actions'
import { boolean, number, select, text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { ButtonActionType } from '@/components/Atoms/Button'
import { NotificationBox } from './NotificationBox'
import {
  AnimationType,
  ArrowPosition,
  ButtonLayout,
  NotificationBoxProps,
  NotificationBoxType,
  NotificationButton,
  NotificationContentSort,
  TooltipData,
} from './NotificationBox.interface'

const animation: AnimationType[] = ['fadeIn', 'fadeOut', 'shake']
const arrowPosition: ArrowPosition[] = ['top-left', 'top-right', 'bottom-left', 'bottom-right']
const types: NotificationBoxType[] = ['alert', 'error', 'info', 'success', 'hot']
const buttonLayout: ButtonLayout[] = ['column', 'row']
const actionButtonTypes: ButtonActionType[] = [
  'ghost',
  'primary',
  'prominent',
  'secondary',
  'inverted',
  'outlined',
]

const knobs = (): NotificationBoxProps => {
  return {
    body: text(
      'Body',
      'Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna.'
    ),
    hasTitleIcon: boolean('Icon', true),
    isClosable: boolean('Closable', true),
    title: text('Title', 'This is the result of your last action!'),
    type: select('Type', types, types[1]),
    animation: select('Animation', animation, undefined),
    buttonLayout: select('Button Layout', buttonLayout, buttonLayout[0]),
  }
}
const buttonKnobs = (): NotificationButton =>
  ({
    actionBtnLabel: text('Button Label', 'Lets do it...'),
    onClick: action('click'),
  } as NotificationButton)

const tooltipKnobs = (): TooltipData => {
  return {
    arrowleft: number('Arrow left', 0),
    arrowPosition: select('Arrow Position', arrowPosition, arrowPosition[0]),
    left: number('Left', 40),
    top: number('Top', 40),
  }
}

storiesOf('Design System/Molecules/Notification Box', module)
  .add('Default', () => <NotificationBox {...knobs()} buttons={[{ ...buttonKnobs() }]} />)
  .add('Default, no button', () => <NotificationBox {...knobs()} />)
  .add('Title only', () => <NotificationBox {...knobs()} body="" isClosable={false} />)
  .add('Body only', () => <NotificationBox {...knobs()} title="" isClosable={false} />)
  .add('Title and button', () => (
    <NotificationBox
      {...knobs()}
      buttons={[{ actionBtnLabel: 'Self Destruct', onClick: action('onClick') }]}
      body=""
    />
  ))
  .add('As Tooltip', () => (
    <NotificationBox
      {...knobs()}
      buttons={[{ actionBtnLabel: 'Self Destruct', onClick: action('onClick') }]}
      body=""
      tooltip={{ ...tooltipKnobs() }}
    />
  ))
  .add('With two buttons', () => (
    <NotificationBox
      {...knobs()}
      buttons={[
        {
          actionBtnLabel: 'Button One',
          actionType: select('Button Type #1', actionButtonTypes, 'secondary'),
          onClick: action('onClick'),
        },
        {
          actionBtnLabel: 'Button Two',
          actionType: select('Button Type #2', actionButtonTypes, 'primary'),
          href: 'https://www.mediamarkt.de',
          target: '_blank',
        },
      ]}
      tooltip={{ ...tooltipKnobs() }}
    />
  ))

  .add('With links', () => (
    <BrowserRouter>
      <NotificationBox
        {...knobs()}
        links={[
          { label: 'Link One', to: 'https://mediamarkt.com/' },
          { label: 'Link Two', onClick: () => {} },
        ]}
        tooltip={{ ...tooltipKnobs() }}
      />
    </BrowserRouter>
  ))

  .add('With links and buttons', () => (
    <BrowserRouter>
      <NotificationBox
        {...knobs()}
        buttons={[
          {
            actionBtnLabel: 'Button',
            actionType: select('Button Type #1', actionButtonTypes, 'secondary'),
            onClick: action('onClick'),
          },
        ]}
        links={[
          {
            label: 'Link',
            to: 'https://mediamarkt.com/',
            target: '_blank',
            decorationColor: 'primary',
            bold: true,
            iconLeft: null,
            inline: false,
            fontSize: 'sm',
          },
        ]}
        sort={select(
          'Sort',
          [NotificationContentSort.LinksButtons, NotificationContentSort.ButtonsLinks],
          NotificationContentSort.ButtonsLinks
        )}
        alignLinks="center"
        tooltip={{ ...tooltipKnobs() }}
      />
    </BrowserRouter>
  ))
