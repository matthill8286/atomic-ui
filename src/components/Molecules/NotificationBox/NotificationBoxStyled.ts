import { Button } from '@/components/Atoms/Button'
import { Icon } from '@/components/Atoms/Icon'
import { fadingIn, fadingOut, shaking } from '@/styles/animation'
import { media } from '@/styles/media'
import { margin } from '@/styles/sc-shared-functions'
import { css, styled } from '@/styles/styled'
import { MarginProps } from '@/types'
import { ButtonLayout, NotificationBoxProps } from './NotificationBox.interface'

export const StyledNotificationHeadline = styled.div`
  display: flex;
  align-items: center;

  ${Icon} {
    margin-right: ${({ theme }) => theme.spacing.base.xs};

    > svg {
      position: relative;

      ${media.md} {
        width: 24px;
        height: 24px;
      }
    }
  }
`

export const StyledCopyTextDiv = styled.div<MarginProps>`
  margin-top: ${({ theme }) => theme.spacing.base.xs};
  ${margin}

  &:first-child {
    &,
    > * {
      margin-top: 0;
    }
  }

  &:last-child > * {
    margin-bottom: 0;
  }
`

export const StyledButton = styled(Button)``

const buttonRowCss = css`
  flex-direction: row;

  & ${StyledButton}:first-child {
    margin-right: ${({ theme }) => theme.spacing.base.xs};
  }

  & ${StyledButton}:last-child {
    margin-left: ${({ theme }) => theme.spacing.base.xs};
  }
`

const buttonColumnCss = css`
  & ${StyledButton}:nth-child(n + 2) {
    margin-top: ${({ theme }) => theme.spacing.base.xs};
  }
`

export const StyledButtonDiv = styled.div<{ buttonLayout: ButtonLayout }>`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  margin-top: ${({ theme }) => theme.spacing.base.md};

  ${({ buttonLayout }) => (buttonLayout === 'row' ? buttonRowCss : buttonColumnCss)}
`

export const StyledLinkDiv = styled.div<{ alignLinks: 'center' | 'left' }>`
  display: flex;
  align-items: start;
  flex-direction: column;
  justify-content: start;
  font-size: ${({ theme }) => theme.font.size.sm};

  ${StyledButtonDiv} + & {
    margin-top: ${({ theme }) => theme.spacing.base.sm};
  }

  ${({ alignLinks }) =>
    alignLinks === 'center' &&
    css`
      > a,
      button {
        margin: 0 auto;
      }
    `}
`

export const StyledNotificationClose = styled.button`
  appearance: none;
  border: 0;
  padding: 0;
  margin: 0;
  border-radius: 0;
  background: transparent;
  position: absolute;
  cursor: pointer;
  right: 0;
  top: 0;
  z-index: 1;
  width: ${({ theme }) => theme.spacing.base.lg};
  height: ${({ theme }) => theme.spacing.base.lg};
  display: flex;
  place-items: center;
  place-content: center;

  &:focus {
    outline: 0;
    user-focus: none;
  }
`

export const StyledNotificationBoxWrapper = styled.div<NotificationBoxProps>`
  ${({ rootPosition }) => `position: ` + (rootPosition ? `${rootPosition};` : `absolute`) + `;`}
  min-width: 0;

  z-index: 25;

  ${media.sm} {
    max-width: ${({ theme }) => 64 * theme.defaultSpacing}px;
    ${({ maxWidth }) => maxWidth && `max-width: ${maxWidth};`}
  }
  ${({ maxWidth }) => maxWidth && `max-width: ${maxWidth};`}

  ${({ animation, title, tooltip }) => [
    animation === 'fadeIn' && fadingIn,
    animation === 'fadeOut' && fadingOut,
    animation === 'shake' && shaking,

    !title &&
      css`
        p:first-child {
          margin-top: 0;
        }
      `,

    tooltip &&
      css`
        top: ${tooltip.top}px;
        left: ${tooltip.left}px;
        width: ${tooltip.width ? `${tooltip.width}px` : 'inherit'};
      `,
  ]}
`

export const StyledArrow = styled.span<{ arrowLeft?: number; arrowPosition: string[] }>`
  position: absolute;
  background-color: ${({ theme }) => theme.color.white};
  border-radius: ${({ theme }) => theme.dimension.borderRadius1};
  border: none;
  height: 20px;
  margin: 0;
  padding: 0;
  transform: rotate(45deg);
  width: 20px;

  ${({ arrowLeft, arrowPosition, theme }) => [
    arrowPosition[0] === 'top'
      ? `top: -${theme.spacing.base.xs};`
      : `bottom: -${theme.spacing.base.xs};`,
    arrowPosition[1] === 'left'
      ? `left: ${theme.spacing.base.md};`
      : `right: ${theme.spacing.base.md};`,
    arrowLeft && `left: ${arrowLeft}px; right: auto;`,
  ]}
`

export const StyledBodyHtml = styled.span`
  color: inherit;

  a,
  a:hover,
  a:focus,
  a:active {
    color: inherit;
  }
`
