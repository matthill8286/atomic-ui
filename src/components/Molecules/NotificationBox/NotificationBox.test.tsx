import React from 'react'
import { renderWithTheme } from '@/testRenderer'
import { NotificationBox } from './NotificationBox'

describe('NotificationBox', () => {
  it('renders corretly', () => {
    const tree = renderWithTheme(
      <NotificationBox
        buttons={[{ actionBtnLabel: 'Lets do it...', onClick: jest.fn() }]}
        body="Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna."
        hasTitleIcon
        isClosable
        onClose={jest.fn()}
        title="This is the result of your last action!"
        type="error"
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
