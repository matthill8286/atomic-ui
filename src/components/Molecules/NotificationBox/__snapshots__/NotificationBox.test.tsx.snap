// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBox renders corretly 1`] = `
.c7 {
  color: #333333;
  font-size: 14px;
  line-height: 20px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c5 {
  display: block;
  box-sizing: border-box;
  line-height: 1.43;
  overflow: hidden;
  padding: 24px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.07),0 2px 1px -1px rgba(0,0,0,0.06),0 1px 3px 0 rgba(0,0,0,0.10);
  color: #000000;
  background-color: #FFFFFF;
  border-radius: 4px;
  height: 100%;
  width: 100%;
}

.c4 {
  position: relative;
  outline: none;
  width: 100%;
}

.c4 .Cardstyled__StyledBadgeLineWrapper-sc-9bh7mr-0 {
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 svg {
  width: 16px;
  height: 16px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #333333;
}

.c3 svg g,
.c3 svg p {
  fill: #333333;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 40px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  width: 100%;
  border: 2px solid #FFFFFF;
  background: #999999;
  color: #FFFFFF;
}

.c12 .c2 {
  padding: 0 8px;
}

.c12 .Status-sc-1moqiqm-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c12:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c12:hover {
  box-shadow: none;
  background: #999999;
}

.c12 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 .c2 {
  margin-right: 8px;
}

.c6 .c2 > svg {
  position: relative;
}

.c8 {
  margin-top: 8px;
}

.c8:first-child,
.c8:first-child > * {
  margin-top: 0;
}

.c8:last-child > * {
  margin-bottom: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 24px;
}

.c11 .NotificationBoxStyled__StyledButton-zwad2t-2:nth-child(n + 2) {
  margin-top: 8px;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  padding: 0;
  margin: 0;
  border-radius: 0;
  background: transparent;
  position: absolute;
  cursor: pointer;
  right: 0;
  top: 0;
  z-index: 1;
  width: 32px;
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  place-items: center;
  place-content: center;
}

.c1:focus {
  outline: 0;
  -webkit-user-focus: none;
  -moz-user-focus: none;
  -ms-user-focus: none;
  user-focus: none;
}

.c0 {
  position: absolute;
  min-width: 0;
  z-index: 25;
}

.c0 p:first-child {
  margin-top: 0;
}

.c10 {
  color: inherit;
}

.c10 a,
.c10 a:hover,
.c10 a:focus,
.c10 a:active {
  color: inherit;
}

@media (min-width:512px) {
  .c4:hover .Cardstyled__StyledBadgeLineWrapper-sc-9bh7mr-0,
  .c4:hover .Cardstyled__StyledFeaturedAssetsContainer-sc-9bh7mr-3 {
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

@media (min-width:752px) {
  .c6 .c2 > svg {
    width: 24px;
    height: 24px;
  }
}

@media (min-width:512px) {
  .c0 {
    max-width: 512px;
  }
}

<div
  className="c0"
  data-test="notification-box"
  type="error"
>
  <button
    className="c1"
    onClick={[MockFunction]}
  >
    <div
      className="c2 c3"
      color="grey6"
      height={16}
      width={16}
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.85 16.44a.5.5 0 010 .71l-.7.7a.5.5 0 01-.71 0L12 13.41l-4.44 4.44a.5.5 0 01-.71 0l-.7-.7a.5.5 0 010-.71L10.59 12 6.15 7.56a.5.5 0 010-.71l.7-.7a.5.5 0 01.71 0L12 10.59l4.44-4.44a.5.5 0 01.71 0l.7.7a.5.5 0 010 .71L13.41 12z"
        />
      </svg>
    </div>
  </button>
  <div
    className="c4"
  >
    <div
      className="c5"
      display="block"
      elevation={1}
      shape="rounded-small"
    >
      <div
        className="c6"
      >
        <div
          className="c2 c3"
          color="grey6"
          height={16}
          width={16}
        >
          <svg
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2a10 10 0 1010 10A10 10 0 0012 2zm0 18a8 8 0 118-8 8 8 0 01-8 8zm3.09-10.56l-.53-.53a.48.48 0 00-.7 0L12 10.76l-1.86-1.85a.48.48 0 00-.7 0l-.53.53a.48.48 0 000 .7L10.76 12l-1.85 1.86a.48.48 0 000 .7l.53.53a.48.48 0 00.7 0L12 13.24l1.86 1.85a.48.48 0 00.7 0l.53-.53a.48.48 0 000-.7L13.24 12l1.85-1.86a.48.48 0 000-.7z"
            />
          </svg>
        </div>
        <div
          className="c7"
          color="grey6"
          data-test="notification-box-title"
          display="flex"
          fontSize="xs"
        >
          This is the result of your last action!
        </div>
      </div>
      <div
        className="c8"
      >
        <p
          className="c9"
          color="grey5"
          data-test="notification-box-body"
          fontSize="sm"
        >
          <span
            className="c10"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna.",
              }
            }
          />
        </p>
      </div>
      <div
        className="c11"
      >
        <button
          className="c12 NotificationBoxStyled__StyledButton-zwad2t-2"
          disabled={false}
          onClick={[Function]}
          size="sm"
          type="button"
        >
          Lets do it...
        </button>
      </div>
    </div>
  </div>
</div>
`;
